var/spack/repos/builtin/packages/root/package.py

```diff
@@ -31,71 +43,431 @@ class Root(CMakePackage):
+    depends_on('freetype')
     depends_on('libpng')
-    depends_on('jpeg')
-    depends_on('gsl')
-    depends_on('python@2.7:')
-    # depends_on('opengl')
-    depends_on('graphviz', when='+graphviz')
-    # depends_on('kerberos')
-    depends_on('libxml2+python')
-    depends_on('openssl')
-    # depends_on('castor')
-    # depends_on('rfio')
-    # depends_on('mysql')
-    # depends_on('oracle')
-    # depends_on('odbc')
-    # depends_on('postgresql')
-    depends_on('sqlite')
-    # depends_on('pythia')
-    depends_on('fftw')
-    depends_on('cfitsio')
+    depends_on('ncurses')
+    depends_on('zlib')
+
+    # X-Graphics
+    depends_on('libx11',  when="+x")
+    depends_on('libxext', when="+x")
+    depends_on('libxft',  when="+x")
+    depends_on('libxpm',  when="+x")
+
+    # OpenGL
+    depends_on('ftgl',  when="+x+opengl")
+    depends_on('glew',  when="+x+opengl")
+    depends_on('gl',    when="+x+opengl")
+    depends_on('glu',   when="+x+opengl")
+    depends_on('gl2ps', when="+x+opengl")
+
+    # Qt4
+    depends_on('qt', when='+qt4')
+
+    # TMVA
+    depends_on('py-numpy', when='+tmva')
+
+    # Asimage variant would need one of these two
+    # For the moment, we use the libafterimage provided by the root sources
+    # depends_on('libafterimage',    when='+asimage') - not supported
+    # depends_on('afterstep@2.2.11', when='+asimage') - not supported
+
+    # Optional dependencies
+    depends_on('avahi',     when='+avahi')
+    depends_on('davix',     when='+davix')
+    depends_on('cfitsio',   when='+fits')
+    depends_on('fftw',      when='+fftw')
+    depends_on('graphviz',  when='+graphviz')
+    depends_on('gsl',       when='+gsl')
+    depends_on('http',      when='+http')
+    depends_on('jemalloc',  when='+jemalloc')
+    depends_on('kerberos',  when='+kerberos')
+    depends_on('ldap',      when='+ldap')
+    depends_on('libcxx',    when='+libcxx')
+    # depends_on('mysql',    when='+mysql')  - not supported
+    depends_on('odbc',      when='+odbc')
+    # depends_on('oracle',   when='+oracle')
+    depends_on('openssl',   when='+ssl')
+    depends_on('postgresql', when='+postgres')
+    depends_on('pythia6@6:6.999+root',  when='+pythia6')
+    depends_on('pythia@8:8.999',  when='+pythia8')
+    depends_on('python@2.7:',     when='+python', type=('build', 'run'))
+    depends_on('r',         when='+r', type=('build', 'run'))
+    depends_on('r-cpp',     when='+r', type=('build', 'run'))
+    depends_on('r-inside',  when='+r', type=('build', 'run'))
+    depends_on('shadow',    when='+shadow')
+    depends_on('sqlite',    when='+sqlite')
+    depends_on('tbb',       when='+tbb')
+    depends_on('unuran',    when='+unuran')
+    depends_on('vc',        when='+vc')
+    depends_on('veccore',   when='+veccore')
+    depends_on('vdt',       when='+vdt')
+    depends_on('libxml2',   when='+xml')
+    depends_on('xrootd',    when='+xrootd')
+    # depends_on('hdfs') - supported (TODO)
+
+    # Old dependencies
```

Comment:
Are these needed for older versions? If so, it may be worthwhile to add a constraint vs. commenting them out:

`depends_on('binutils', when='@:x.y')` (all versions up to x.y need binutils)

If not, I think they can be removed.

Reply:
I think this block of _old_ dependencies can be removed:

```
    # depends_on('binutils')
    # depends_on('gif')  - provided by builtin afterimage
    # depends_on('jpeg') - provided by builtin afterimage
    # depends_on('libice')
    # depends_on('libsm')
    # depends_on('libxml2+python')
```

My initial intention was to check if anyone was against removing them since the current version of this recipe uses these packages. However, now most of them are provided by other packages previously defined as a dependency, module `libxml2+python` which I don't know what was needed for. I will remove this block.

On the contrary, I would keep the following two blocks of commented dependencies ("not supported" and "Grid packages"). Although very rarely someone may need them, they are still dependencies that could be added for completeness (not as part of this PR).

Reply:
> On the contrary, I would keep the following two blocks of commented dependencies ("not supported" and "Grid packages").

Sounds good - I was only talking about the "old" section. Commented dependencies in general are fine (and you have explanatory comments by all of them).